account_cutoff_base/models/account_cutoff.py

```diff
@@ -0,0 +1,358 @@
+# © 2013-2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class AccountCutoff(models.Model):
+    _name = 'account.cutoff'
+    _rec_name = 'cutoff_date'
+    _order = 'cutoff_date desc'
+    _inherit = ['mail.thread']
+    _description = 'Account Cut-off'
+
+    @api.multi
+    @api.depends('line_ids', 'line_ids.cutoff_amount')
+    def _compute_total_cutoff(self):
+        for cutoff in self:
+            tamount = 0.0
+            for line in cutoff.line_ids:
+                tamount += line.cutoff_amount
+            cutoff.total_cutoff_amount = tamount
+
+    @api.model
+    def _default_move_label(self):
+        type = self._context.get('type')
+        label = ''
+        if type == 'accrued_expense':
+            label = _('Accrued Expense')
+        elif type == 'accrued_revenue':
+            label = _('Accrued Revenue')
+        elif type == 'prepaid_revenue':
+            label = _('Prepaid Revenue')
+        elif type == 'prepaid_expense':
+            label = _('Prepaid Expense')
+        return label
+
+    @api.model
+    def _inherit_default_cutoff_account_id(self):
+        """Function designed to be inherited by other cutoff modules"""
+        return None
+
+    @api.model
+    def _default_cutoff_account_id(self):
+        """This function can't be inherited, so we use a second function"""
+        return self._inherit_default_cutoff_account_id()
+
+    cutoff_date = fields.Date(
+        string='Cut-off Date', readonly=True,
+        states={'draft': [('readonly', False)]}, copy=False,
+        track_visibility='onchange')
+    type = fields.Selection([
```

Comment:
would be better to rename this field to `cutoff_type`, according also to another model below. You can use `oldname='type'`